### PR TITLE
src/mon: clarify message for pool creation failure due to max_pgs_per_osd 

### PR DESCRIPTION
**Pull request by anthonyeleven**
src/mon: clarify message for pool creation failure due to max_pgs_per_osd

This was inspired by the `Re: [ceph-users] proxmox hyperconverged pg calculations in ceph pacific, pve 7.2` thread.

The current message sort of mixes up up `max_pgs_per_osd` with `mon_max_pool_pg_num`.

Fixes: https://tracker.ceph.com/issues/58149
Signed-off-by: Anthony D'Atri <anthony.datri@gmail.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by ljflores — APPROVED** [review]
Makes sense to me!

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -7938,15 +7938,16 @@ int OSDMonitor::check_pg_num(int64_t pool, int pg_num, int size, int crush_rule,
...
     }
     *ss << " pg_num " << pg_num << " size " << size
-	<< " would mean " << projected
-	<< " total pgs, which exceeds max " << max_pgs
-	<< " (mon_max_pg_per_osd " << max_pgs_per_osd
-	<< " * num_in_osds " << num_osds << ")";
+	<< " for this new pool would result in " << projected_pgs_per_osd
+	<< " cumulative PGs per OSD (" << projected
+	<< " total PG replicas on " << num_osds
+	<< " 'in' OSDs) which exceeds the mon_max_pg_per_osd value of " << max_pgs_per_osd;
```
Comment (ljflores) [inline src/mon/OSDMonitor.cc:7950]: Minor nit: I personally don't think "cumulative" is necessary here.
```suggestion
	<< " for this new pool would result in " << projected_pgs_per_osd
	<< " PGs per OSD (" << projected
	<< " total PG replicas on " << num_osds
	<< " 'in' OSDs) which exceeds the mon_max_pg_per_osd value of " << max_pgs_per_osd;
```

Comment (anthonyeleven) [reply src/mon/OSDMonitor.cc:7950]: I get what your mean, FWIW I put "cumulative" in there to make it clear that this number reflects existing pools too, not just the new pool to be created.

Comment (ljflores) [reply src/mon/OSDMonitor.cc:7950]: Oh, I see. In that case, leave it in!

Comment (kamoltat) [reply src/mon/OSDMonitor.cc:7950]: Yea that make sense to me too +1

**Review by kamoltat — APPROVED** [review]

**Comment by kamoltat** [comment]
jenkins test make check arm64

**Comment by kamoltat** [comment]
jenkins test windows

**Comment by anthonyeleven** [comment]
FWIW I'm receiving dozens of `Run failed: Create backport trackers for trackers in "Pending Backport" state - main (6870298)` email messages each day

**Comment by anthonyeleven** [comment]
> @anthonyeleven did you fork the ceph repository recently? If so, it clones the github actions and sometimes sends emails about them. You can go to https://github.com/anthonyeleven/ceph/settings/actions and hit "Disable actions", or possibly modify your notification preferences.

Thanks.  My fork has been around for a while, I did just clone to two new local systems.  I'll give the above a shot - this did not happen to me previously.

**Comment by ljflores** [comment]
Rados suite review: https://pulpito.ceph.com/?branch=wip-yuri3-testing-2022-12-06-1211

Failures, unrelated:
    1. https://tracker.ceph.com/issues/57311
    2. https://tracker.ceph.com/issues/58098
    3. https://tracker.ceph.com/issues/58096
    4. https://tracker.ceph.com/issues/52321
    5. https://tracker.ceph.com/issues/57731
    6. https://tracker.ceph.com/issues/57546

Details:
    1. rook: ensure CRDs are installed first - Ceph - Orchestrator
    2. qa/workunits/rados/test_crash.sh: crashes are never posted - Ceph - RADOS
    3. test_cluster_set_reset_user_config: NFS mount fails due to missing ceph directory - Ceph - Orchestrator
    4. qa/tasks/rook times out: 'check osd count' reached maximum tries (90) after waiting for 900 seconds - Ceph - Orchestrator
    5. Problem: package container-selinux conflicts with udica < 0.2.6-1 provided by udica-0.2.4-1 - Infrastructure
    6. rados/thrash-erasure-code: wait_for_recovery timeout due to "active+clean+remapped+laggy" pgs - Ceph - RADOS

**Comment by anthonyeleven** [comment]
Thanks!    Mixed feelings about potential backports, it'd be nice to be more clear for people running prior releases, but I'm hesitant to introduce a user-visible (and automation-visible) change in a minor release.

**Pull request merged**